### PR TITLE
Add helm version to toCreate on update

### PR DESCRIPTION
**Pull request by brendarearden**
**Problem**
When a new version is added to a helm 3 catalog that is already in rancher, update is triggered instead of create, so the new version is not getting the helm version assigned to it

**Solution**
Add helm version to the update method to ensure new versions created on update have the helm version assigned

**Issue** 
#27252

### CODE REVIEW

File: tests/integration/suite/test_app.py [hunk 1]
```diff
@@ -647,6 +647,92 @@ def test_app_has_helmversion(admin_pc, admin_mc, remove_resource):
...
+    # catalog
+    catalog_data = {
+      'name': catalog_name,
+      'branch': "helmversion-onupdate-2v",
+      'url': "https://github.com/brendarearden/integration-test-charts",
+      'helmVersion': "helm_v3"
+    }
+    helm3_catalog = catalog_client.update(helm3_catalog, catalog_data)
+
+    def ensure_udpated_catalog(catalog):
```
Comment (rmweir) [inline tests/integration/suite/test_app.py:696]: udpated -> updated

**Comment by batiati** [comment]
Fixes #26920

File: tests/integration/suite/test_app.py [hunk 2]
```diff
@@ -647,6 +647,92 @@ def test_app_has_helmversion(admin_pc, admin_mc, remove_resource):
...
+    catalog_client = admin_mc.client
+    catalog_name = random_str()
+    app1_name = random_str()
+    app2_name = random_str()
+    cat_base = "catalog://?catalog=" + catalog_name +\
+        "&template=rancher-v3-issue&version="
+    helm3_catalog = catalog_client.create_catalog(
+      name=catalog_name,
+      branch="helmversion-onupdate-1v",
+      url="https://github.com/brendarearden/integration-test-charts",
```
Comment (dramich) [inline tests/integration/suite/test_app.py:663]: This needs to live under a rancher repo

File: tests/integration/suite/test_app.py [hunk 3]
```diff
@@ -647,6 +647,92 @@ def test_app_has_helmversion(admin_pc, admin_mc, remove_resource):
...
+        templatesString = ','.join([str(i) for i in templates])
+        if "0.1.1" in templatesString:
+            return catalog
+        return None
+    helm3_catalog = wait_for(
+      lambda: ensure_updated_catalog(helm3_catalog),
+      fail_handler=lambda:
+      "Timed out waiting for catalog to stop transitioning")
+    project_client = user_project_client(admin_pc, admin_pc.project)
+    # update existing app with new version to ensure correct
```
Comment (dramich) [inline tests/integration/suite/test_app.py:708]: Ideally the template should be checked as that is really our source of truth, the app could be holding old data from pre upgrade thus giving a false positive.

Comment (dramich) [reply tests/integration/suite/test_app.py:708]: Both can be checked and this also tests the upgrade in general.

**Pull request merged**